### PR TITLE
Add feature for multiple lattice parameters/phases in polycrystal_from_odf

### PR DESCRIPTION
**Pull request by vamatya**
Feature add for ingesting multiple lattice parameters, sgname and cif files, and corresponding phase fractions for `polycrystal_from_odf` function.

### CODE REVIEW

File: xrd_simulator/templates.py [hunk 1]
```diff
@@ -168,10 +171,18 @@ def polycrystal_from_odf(
     )
 
     mesh = TetraMesh._build_tetramesh(cylinder)
-
-    # Sample is uniformly single phase
-    phases = [Phase(unit_cell, sgname, path_to_cif_file)]
-    element_phase_map = np.zeros((mesh.number_of_elements,)).astype(int)
+
```
Comment (AxelHenningsson) [inline xrd_simulator/templates.py:174]: This looks solid to me. :=)

Altough, we need [to make the unit tests pass](https://github.com/FABLE-3DXRD/xrd_simulator/blob/21e58b6b3f83bb174046416791e057b92b459e68/tests/test_templates.py#L70) / add one for the case of multi phase.

**Review by AxelHenningsson — COMMENTED** [review]
This looks solid to me. :=)

Altough, we need [to make the unit tests pass](https://github.com/FABLE-3DXRD/xrd_simulator/blob/21e58b6b3f83bb174046416791e057b92b459e68/tests/test_templates.py#L70) / add one for the case of multi phase, and update the examples.

**Comment by AxelHenningsson** [comment]
Great that you are contributing @vamatya ❤️ 🤟 🎉 

Some things still need to be done before we can merge.

1. Make the unit tests pass - i.e. propagate any changes into the unit tests.
2. Add a simple unit tests that verify that the new feature works as expected.
3. Update the examples to reflect the changes - and to showcase your new and awesome feature that you added ! 🚀 

Cheers,
Axel

File: xrd_simulator/templates.py [hunk 2]
```diff
@@ -131,6 +132,8 @@ def polycrystal_from_odf(
             strain_tensor(x) -> :obj:`numpy array` of shape ``(3,3)`` where input variable ``x`` is
             a :obj:`numpy array` of shape ``(3,)`` representing a spatial coordinate in the
             cylinder (x,y,z).
+        phase_fraction (:obj: `list` of :obj:`float`): Phase fraction represented as probability, summing up to 1.
+            Default None; will take even distribution of crystals.
 
     Returns:
         (:obj:`xrd_simulator.polycrystal.Polycrystal`)
```
[the remark this answers is not in the document]
Comment (vamatya) [reply xrd_simulator/templates.py:146]: I added a 2 phase example, albeit with trivial input values.

File: xrd_simulator/templates.py [hunk 3]
```diff
@@ -117,11 +118,11 @@ def polycrystal_from_odf(
              microns.
         sample_bounding_cylinder_radius (:obj:`float`): Radius of sample cylinder in units of
             microns.
-        unit_cell (:obj:`list` of :obj:`float`): Crystal unit cell representation of the form
-            [a,b,c,alpha,beta,gamma], where alpha,beta and gamma are in units of degrees while
+        unit_cell (:obj:`list of lists` of :obj:`float`): Crystal unit cell representation of the form
+            [[a,b,c,alpha,beta,gamma],], where alpha,beta and gamma are in units of degrees while
             a,b and c are in units of anstrom.
```
[the remark this answers is not in the document]
Comment (vamatya) [reply xrd_simulator/templates.py]: This is updated.

**Pull request merged**